File: lessons/lesson-03-strings-wrappers/java-code/Strings.java
// Java — Strings

public class Strings {
    public static void main(String[] args) {

        String s = "Hello World";

        System.out.println(s.length());          // 11 — METHOD (parentheses!)
        System.out.println(s.toUpperCase());     // "HELLO WORLD"
        System.out.println(s.toLowerCase());     // "hello world"
        System.out.println(s.substring(0, 5));   // "Hello"
        System.out.println(s.indexOf("World"));  // 6
        System.out.println(s.contains("llo"));   // true (not "includes"!)
        System.out.println(s.trim());

        // Split returns String array (not a dynamic JS array)
        String[] words = s.split(" ");
        System.out.println(words[0]); // "Hello"
        System.out.println(words[1]); // "World"

        // No template literals — use concatenation
        String name = "Alice";
        System.out.println("Hello, " + name + "!");

        // Or printf
        System.out.printf("Hello, %s!%n", name);

        // String equality — USE .equals() !!
        String a = "hello";
        String b = "hello";
        System.out.println(a == b);       // might be true (string pool quirk)
        System.out.println(a.equals(b));  // ALWAYS true — use this!

        // charAt — get a single character
        char first = s.charAt(0);  // 'H'
        System.out.println("First char: " + first);

        // Wrapper classes — parsing
        int num = Integer.parseInt("42");       // like parseInt()
        double dec = Double.parseDouble("3.14"); // like parseFloat()
        String str = String.valueOf(42);         // like String(42)
        System.out.println(num + " " + dec + " " + str);

        // Character utility methods (no JS equivalent)
        System.out.println(Character.isLetter('A'));   // true
        System.out.println(Character.isDigit('5'));    // true
        System.out.println(Character.toUpperCase('a')); // 'A'
    }
}

File: lessons/lesson-03-strings-wrappers/js-compare/strings.js
// JavaScript — Strings
// Run: node strings.js

let s = "Hello World";

console.log(s.length);           // 11 — property (no parentheses)
console.log(s.toUpperCase());    // "HELLO WORLD"
console.log(s.toLowerCase());    // "hello world"
console.log(s.substring(0, 5)); // "Hello"
console.log(s.indexOf("World"));// 6
console.log(s.includes("llo")); // true
console.log(s.split(" "));     // ["Hello", "World"]
console.log(s.trim());

// Template literals
let name = "Alice";
console.log(`Hello, ${name}!`);

// Equality
let a = "hello";
let b = "hello";
console.log(a === b);  // true

// Parsing
console.log(parseInt("42"));     // 42
console.log(parseFloat("3.14")); // 3.14
console.log(String(42));         // "42"
